migration/migrator/data/submitty_db.sql

```diff
@@ -441,6 +441,8 @@ CREATE TABLE public.courses (
     status smallint DEFAULT 1 NOT NULL,
     group_name character varying(255) NOT NULL,
     owner_name character varying(255) NOT NULL,
+    self_registration_type smallint DEFAULT 0,
+    default_section_id character varying(255),
```

Comment:
You could also add a foreign key constraint to this to make sure the section exists.

Reply:
Is it possible to have a foreign key linked across different databases?

Reply:
Ah, good point.  Foreign key constraints aren't supported across databases unfortunately.  You could possibly set up a trigger to mimic the behavior if you think it would be useful though.

Reply:
Having a trigger could be useful because it could enforce self registration getting turned off automatically if the referenced section is removed.

Reply:
Would it be easier to do this, or to not allow the section to be removed until self registration is disabled (via the controllers)?

Reply:
Having a trigger is not mutually exclusive with having checks in a controller somewhere.  Checks in a controller should in theory handle everything, but having a trigger acting as a constraint would add an additional level of data integrity protection in case something unexpected happens with the PHP code, either from your changes or down the line when some brand new inexperienced dev comes along and doesn't see the implications of a change they make.

Reply:
After discussion with Barb, this isn't a priority issue, while it may 'break' something if the database gets in a bad state, it is a lower priority thing due to most users using the UI and not manually editing the database.